cli/azd/resources/alpha_features.yaml

```diff
@@ -1,5 +1,3 @@
-- id: terraform
```

Comment:
Hopefully such removals aren't breaking i.e., us removing alpha features and promoting to beta doesn't cause any errors when external templates/pipelines trying to enable/disable.

Reply:
For the way how `azd config set` works, there won't be breaking changes.
After this PR, someone calling `azd config set alpha.terraform on` would still adding that configuration to azd config, but that's ignored by azd.  Similar to adding any other config key/value pair like `alpha.foo bar`